src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -863,10 +967,40 @@ private RuntimeExecutionResults ProcessCore()
                 this.dataSourcesToProcess);
 
             var executors = CreateExecutors(allDataSourceAssociations);
+
+            var extendedTables = new HashSet<ITableExtensionReference>();
+            var processorTables = new Dictionary<TableDescriptor, ICustomDataProcessor>();
+
+            foreach (var table in this.enabledTables)
+            {
+                if (this.extensionRoot.TablesById.TryGetValue(table.Guid, out ITableExtensionReference reference))
+                {
+                    extendedTables.Add(reference);
+                }
+                else
+                {
+                    foreach (var executor in executors.Where(x => x.Context.CustomDataSource.AvailableTables.Contains(table)))
+                    {
+                        executor.Processor.EnableTable(table);
+                        processorTables.Add(table, executor.Processor);
```

Comment:
based on what I remember @zachnew-msft said today... an executor is tied to a CDS, and a table should not exist in multiple custom data sources, so I think semantically it would make more sense to do a .First() instead of a foreach.

clearly not a functional change, just seems more clear to me what we actually expect to happen if we did it that way

Reply:
Tweaked it to be "Single" since it should be there